### PR TITLE
fix: Replace link-input with link-field and social-link-field

### LINKED ISSUE #5570: Replace link-input with link-field and social-link-field
Replace all link-input component calls with link-field and social-link-field and then remove the component and all segmented link shenanigans from the project

### PR DESCRIPTION
**Pull request by daretobedifferent18**
Fixes #5570 

#### Short description of what this resolves:
Replace all link-input component calls with link-field and social-link-field and then remove the component and all segmented link shenanigans from the project

#### TODO

- [x] Replace LinkInput with LinkField or SocialLinkField
- [x] Remove Segmented Links
- [x] Remove Transation for LinkInput
- [x] Take care of styling

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by daretobedifferent18** [comment]
@iamareebjamal Please let me know that should I remove segmented links where they where never being used also?
I have refactored almost everything.
Please see the changes done.

File: app/templates/components/forms/admin/content/social-links-form.hbs [hunk 1]
```diff
@@ -1,65 +1,66 @@
...
   <div class="field">
     <label>{{t 'Twitter'}}</label>
-    <Widgets::Forms::LinkInput
-      @segmentedLink={{this.socials.segmentedTwitterUrl}}
-      @linkName="twitter"
-      @isSocialLink={{true}}
-      @inputId="twitter" />
+    <Widgets::Forms::SocialLinkField
+      @inputId={{'twitter'}}
+      @site={{'twitter'}}
```
Comment (iamareebjamal) [inline app/templates/components/forms/admin/content/social-links-form.hbs:13]: No need wrap this in {{}}

File: app/components/widgets/forms/social-link-field.hbs [hunk 2]
```diff
@@ -1,4 +1,5 @@
-<div class="fields">
+<div class="{{if @addLinks 'fields'}}">
+  {{#if @addLinks}}
```
Comment (iamareebjamal) [inline app/components/widgets/forms/social-link-field.hbs:2]: Why addLinks?

**Comment by iamareebjamal** [comment]
Yes, remove all segmented links and also the helper code

File: app/components/widgets/forms/social-link-field.hbs (hunk 2 above)
Comment (daretobedifferent18) [reply app/components/widgets/forms/social-link-field.hbs:2]: ofcourse we could give this more sensible name but this is bcz at some places we dont need that UiDropdrop and input defined in social-link-field

Comment (iamareebjamal) [reply app/components/widgets/forms/social-link-field.hbs:2]: If no dropdown is needed, why not use LinkField instead of SocialLinkField?

Comment (daretobedifferent18) [reply app/components/widgets/forms/social-link-field.hbs:2]: then I have to manually provide prefix value for each of them for e.g
```@prefix={{'https://twitter.com/'}}```

Comment (daretobedifferent18) [reply app/components/widgets/forms/social-link-field.hbs:2]: If this is ok then I can do like this

Comment (iamareebjamal) [reply app/components/widgets/forms/social-link-field.hbs:2]: Just change prop to `fixed` and if fixed is true, dropdown will not be shown

Comment (daretobedifferent18) [reply app/components/widgets/forms/social-link-field.hbs:2]: rename addLinks to fixed?

Comment (iamareebjamal) [reply app/components/widgets/forms/social-link-field.hbs:2]: Invert it

**Comment by daretobedifferent18** [comment]
@iamareebjamal Please have a look

File: app/components/widgets/forms/social-link-field.hbs [hunk 3]
```diff
@@ -1,4 +1,5 @@
-<div class="fields">
+<div class="{{if @fixed '' 'fields'}}">
+  {{#if (not @fixed)}}
```
Comment (iamareebjamal) [inline app/components/widgets/forms/social-link-field.hbs:2]: unless

File: app/components/widgets/forms/social-link-field.hbs [hunk 4]
```diff
@@ -1,4 +1,5 @@
...
   <div class="three wide field">
```
Comment (iamareebjamal) [inline app/components/widgets/forms/social-link-field.hbs:3]: Change indentation

File: app/models/session.js [hunk 5]
```diff
@@ -59,10 +59,10 @@ export default class Session extends ModelBase.extend({
...
   endsAtTime  : computedDateTimeSplit.bind(this)('endsAt', 'time'),
 
-  segmentedLinkSlidesUrl : computedSegmentedLink.bind(this)('slidesUrl'),
-  segmentedLinkAudioUrl  : computedSegmentedLink.bind(this)('audioUrl'),
-  segmentedLinkVideoUrl  : computedSegmentedLink.bind(this)('videoUrl'),
-  segmentedLinkSignUpUrl : computedSegmentedLink.bind(this)('signUpUrl'),
+  // segmentedLinkSlidesUrl : computedSegmentedLink.bind(this)('slidesUrl'),
+  // segmentedLinkAudioUrl  : computedSegmentedLink.bind(this)('audioUrl'),
+  // segmentedLinkVideoUrl  : computedSegmentedLink.bind(this)('videoUrl'),
+  // segmentedLinkSignUpUrl : computedSegmentedLink.bind(this)('signUpUrl'),
```
Comment (iamareebjamal) [inline app/models/session.js:65]: No need to comment code, delete it

**Comment by daretobedifferent18** [comment]
@iamareebjamal Any other changes to make

File: app/components/widgets/forms/social-link-field.hbs [hunk 6]
```diff
@@ -1,6 +1,7 @@
-<div class="fields">
-  <div class="three wide field">
-    <div class="ui labeled input">
+<div class="{{if @fixed '' 'fields'}}">
```
Comment (iamareebjamal) [inline app/components/widgets/forms/social-link-field.hbs:1]: unless will be better here as well

File: tests/integration/components/widgets/forms/social-link-field-test.ts [hunk 7]
```diff
@@ -17,7 +17,7 @@ module('Integration | Component | widgets/forms/social-link-field', function(hoo
     await render(hbs`{{widgets/forms/social-link-field site='twitter'}}`);
     assertContains(assert, this.element.textContent?.trim(), 'https://twitter.com/');
 
-    await render(hbs`{{widgets/forms/social-link-field site='instagram'}}`);
-    assertContains(assert, this.element.textContent?.trim(), 'https://instagram.com/');
+    await render(hbs`{{widgets/forms/social-link-field site='website'}}`);
```
Comment (iamareebjamal) [inline tests/integration/components/widgets/forms/social-link-field-test.ts:20]: Please revert

Comment (daretobedifferent18) [reply tests/integration/components/widgets/forms/social-link-field-test.ts:20]: Done

File: tests/integration/components/widgets/forms/link-input-test.js [hunk 8]
```diff
@@ -1,13 +0,0 @@
-import { hbs } from 'ember-cli-htmlbars';
-import { module, test } from 'qunit';
-import { setupIntegrationTest } from 'open-event-frontend/tests/helpers/setup-integration-test';
-import { render } from '@ember/test-helpers';
-
-module('Integration | Component | widgets/forms/link input', function(hooks) {
-  setupIntegrationTest(hooks);
-
-  test('it renders', async function(assert) {
-    await render(hbs`{{widgets/forms/link-input}}`);
```
Comment (iamareebjamal) [inline tests/integration/components/widgets/forms/link-input-test.js:10]: Why deleted this? Revert

**Comment by daretobedifferent18** [comment]
should I not delete linkinput comp files??

**Comment by iamareebjamal** [comment]
Commented by mistake. Thought its link-field test

**Comment by daretobedifferent18** [comment]
Any other changes to make

**Comment by iamareebjamal** [comment]
Will test and merge tomorrow

**Comment by iamareebjamal** [comment]
@daretobedifferent18 Please check your gitter

**Comment by daretobedifferent18** [comment]
> @daretobedifferent18 Please check your gitter

ok

**Comment by iamareebjamal** [comment]
@maze-runnar Please review and test locally that everything is working as expected

**Comment by maze-runnar** [comment]
> @maze-runnar Please review and test locally that everything is working as expected

ok

**Comment by maze-runnar** [comment]
it's working fine, but ui field in mobile view -
scrnli_11_24_2020_3-59-25 PM
scrnli_11_24_2020_4-00-48 PM

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 9]
```diff
@@ -71,20 +71,20 @@
...
-      @segmentedLink={{this.data.event.segmentedLiveStreamUrl}}
-      @canRemoveItem={{false}} />
-    <Widgets::Forms::LinkInput
-      @hasLinkName={{true}}
-      @fixedName={{true}}
-      @linkName={{t "Webinar URL"}}
-      @inputId="webinar_url"
-      @segmentedLink={{this.data.event.segmentedWebinarUrl}}
-      @canRemoveItem={{false}} />
+    <div class="six wide field">
```
Comment (maze-runnar) [inline app/templates/components/forms/wizard/basic-details-step.hbs:74]: use` {{unless this.device.isMobile 'six wide'}} field`

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 10]
```diff
@@ -71,20 +71,20 @@
...
+      <label>{{t "Live Stream URL"}}</label>
+      <Widgets::Forms::LinkField
+        @inputId="live_stream_url"
+        @value={{this.data.event.liveStreamUrl}}
+        @onChange={{action (mut this.data.event.liveStreamUrl)}} />
+    </div>
+    <div class="six wide field">
```
Comment (maze-runnar) [inline app/templates/components/forms/wizard/basic-details-step.hbs:81]: here to

**Comment by daretobedifferent18** [comment]
Screenshot from 2020-11-24 20-40-11

**Comment by daretobedifferent18** [comment]
@iamareebjamal I have made all the changes asked for.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**